### PR TITLE
Fix if-not-exists behavior of create index statements

### PR DESCRIPTION
**Pull request by czpmango**
## What type of PR is this?
- [x] bug
- [ ] feature
- [ ] enhancement

#### Issue(s) number: 
https://github.com/vesoft-inc/nebula/issues/4665

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [x] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [x] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

File: src/meta/processors/index/CreateEdgeIndexProcessor.cpp [hunk 1]
```diff
@@ -96,7 +97,15 @@ void CreateEdgeIndexProcessor::process(const cpp2::CreateEdgeIndexReq& req) {
     }
 
     if (checkIndexExist(fields, item)) {
-      resp_.code_ref() = nebula::cpp2::ErrorCode::E_EXISTED;
+      if (ifNotExists) {
+        resp_.code_ref() = nebula::cpp2::ErrorCode::SUCCEEDED;
+        cpp2::ID thriftID;
```
Comment (SuperYoko) [inline src/meta/processors/index/CreateEdgeIndexProcessor.cpp:102]: maybe use tools like  `resp_.id_ref() = to(edgeIndex, EntryType::INDEX);` ?

File: src/meta/processors/index/CreateEdgeIndexProcessor.cpp [hunk 2]
```diff
@@ -96,7 +97,15 @@ void CreateEdgeIndexProcessor::process(const cpp2::CreateEdgeIndexReq& req) {
...
+        // Fill index id to avoid broken promise
+        thriftID.index_id_ref() = item.get_index_id();
+        resp_.id_ref() = thriftID;
```
Comment (panda-sheep) [inline src/meta/processors/index/CreateEdgeIndexProcessor.cpp:105]: It feels a little weird this way
1. The first `ifNotExists` in line 45 is to judge whether the index name exists or not.

2 This place is to judge whether there are other indexes that contain this fields
for example
```
create edge e(a int, b int);
create edge index e_index on e(a, b)
create edge index if not exists e_index_2 on e(a);
```
This returns a successful creation and returns the indexId of the first index.
There is actually only one index e_index

Comment (panda-sheep) [reply src/meta/processors/index/CreateEdgeIndexProcessor.cpp:105]: It shows that the index creation is successful, but when show, the index name does not exist

Comment (czpmango) [reply src/meta/processors/index/CreateEdgeIndexProcessor.cpp:105]: Indeed, this is the point just as you said. 
IMO, index name is a redundant design. 
A better syntax might be this: 
```
create edge index on e(a);
create edge index if not exists on e(a);
rebuild edge index on e(a);
```
In this way, the implementation layer does not need to verify the index name(line 45).
Considering compatibility, this pr is not going to do major refactoring.

File: src/meta/processors/index/CreateEdgeIndexProcessor.cpp (hunk 1 above)
Comment (czpmango) [reply src/meta/processors/index/CreateEdgeIndexProcessor.cpp:102]: Just extract the id from the index item.

File: src/meta/processors/index/CreateEdgeIndexProcessor.cpp (hunk 2 above)
Comment (panda-sheep) [reply src/meta/processors/index/CreateEdgeIndexProcessor.cpp:105]: @czpmango @MuYiYong How is our GQL version index implemented?

Comment (czpmango) [reply src/meta/processors/index/CreateEdgeIndexProcessor.cpp:105]: We should try to avoid similar problems in GQL version.

Comment (panda-sheep) [reply src/meta/processors/index/CreateEdgeIndexProcessor.cpp:105]: 5.0 index design needs to consider this problem, thx @MuYiYong

**Review by panda-sheep — APPROVED** [review]

**Review by SuperYoko — APPROVED** [review]

**Comment by jinyingsunny** [comment]
checked on 3.3-rc

**Pull request merged**